### ISSUE TITLE
Cannot sync over Kobo API

### ISSUE THREAD

**Issue by jvacek** [janeczku/calibre-web#2783]
**Describe the bug/problem**
CW gives me a 500 whenn trying to access the kobo `v1/library/sync` endpoint. This happens for a new non-admin user I created. The library currently has ~580 titles with multiple versions under each. The admin user's token works fine.

**To Reproduce**
1. Create new user
2. Create kobo token
3. Access the endpoint via web browser

**Logfile**
```
[2023-05-15 09:42:39,798] ERROR {cps:1449} Exception on /kobo/85899bdd5c08c7bed2f8cf6a42cc8a79/v1/library/sync [GET]
Traceback (most recent call last):
  File "/opt/.venv/calibreweb/lib/python3.8/site-packages/flask/app.py", line 2073, in wsgi_app
    response = self.full_dispatch_request()
  File "/opt/.venv/calibreweb/lib/python3.8/site-packages/flask/app.py", line 1519, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/opt/.venv/calibreweb/lib/python3.8/site-packages/flask/app.py", line 1517, in full_dispatch_request
    rv = self.dispatch_request()
  File "/opt/.venv/calibreweb/lib/python3.8/site-packages/flask/app.py", line 1503, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "/opt/calibreweb/cps/kobo_auth.py", line 168, in inner
    return f(*args, **kwargs)
  File "/opt/calibreweb/cps/web.py", line 122, in inner
    return f(*args, **kwargs)
  File "/opt/calibreweb/cps/kobo.py", line 224, in HandleSyncRequest
    "BookMetadata": get_metadata(book.Books),
  File "/opt/calibreweb/cps/kobo.py", line 465, in get_metadata
    if get_epub_layout(book, book_data) == 'pre-paginated':
  File "/opt/calibreweb/cps/epub.py", line 58, in get_epub_layout
    p = tree.xpath('/pkg:package/pkg:metadata', namespaces=ns)[0]
IndexError: list index out of range
```
**Expected behavior**
Working endpoint that doesn't 500

**Screenshots**
N/A

**Environment (please complete the following information):**
 - OS: Ubuntu 20.04
 - Python version: Python3.8.10
 - Calibre-Web version: 0.6.20
 - Docker container: None:
 - Special Hardware: N/A
 - Browser: Safari, Firefox

**Additional context**

**Comment by DwightM66** [comment]
I have the exact same error. I have also tried the latest nightly release with the same error.

**Comment by OzzieIsaacs** [comment]
Please check #2840

**Comment by DwightM66** [comment]
After installing the nightly that fixed #[2840](https://github.com/janeczku/calibre-web/issues/2840), I was still getting an error. I then installed the "KePub Output" plugin in Calibre, converted all of my ebooks to KePub format using this. Now, the sync in Calibre-Web is working again, even with a larger library.

**Comment by DwightM66** [comment]
> 

@ZippierOstrich1 I did have the Calibre Web kepubify converter installed, but still had problems. It was not until I used the "KePub Output" plugin in Calibre to convert all of my books to kepub format that sync started working again. My problems started as my library grew over a certain size. Before I hit that threshold size, the kepubify converter seemed to work for sync.  I can't recall what the library size was before things started having problems, but I think it was somewhere around 900 ebooks.
